doc/pxe.md

```diff
@@ -0,0 +1,132 @@
+# Using PXE and Iguana
+
+This document explains how to run D-Installer on PXE with the help of Iguana. The described setup
+uses libvirt, but you can adapt the overall approach to other scenarios (like running your TFTP
+server).
+
+Additionally, it offers some helpful tips for debugging D-Installer problems.
+
+## Set up
+
+The process can be summarized in these steps:
+
+1. Set up the TFTP tree, defining a boot option for D-Installer + Iguana.
+2. Configure libvirt network to serve the tree.
+3. Prepare the initial ramdisk image (initrd), based on Iguana.
+4. Boot from PXE.
+
+### Set up the TFTP tree
+
+The TFTP tree should contain the SYSLINUX boot loader. You can copy the required files from the
+`syslinux` package.
+
+    zypper in syslinux
+    mkdir /srv/tftpboot
+    cp /usr/share/syslinux/pxelinux.0 /srv/tftpboot
+    mkdir /srv/tftpboot/pxelinux.cfg
+
+To define a boot option to run D-Installer, add a `/srv/tftpboot/pxelinux.cfg/default` file with the
+following content:
+
+```
+default iguana
+
+label iguana
+  ipappend 2
+  kernel vmlinuz-iguana
+  append initrd=initrd-iguana rd.iguana.control_url=tftp://192.168.122.1/d-installer.yaml rd.iguana.debug=1
+
+display		message
+implicit	1
+prompt		1
+timeout		50
+```
+
+Do not worry about the kernel, the initrd or the `d-installer.yaml` file, we will jump into it
+later.
+
+### Configure libvirt to server TFTP files
+
+To instruct libvirt to serve the TFTP files, you must add the `tftp` and `bootp` elements to the
+network configuration. Use the `virsh net-edit default` command to edit the configuration and adapt
+it accordingly. Here is an example:
+
+```xml
+<network connections='1'>
+  <name>default</name>
+  <uuid>639e02a7-fcbd-4cf3-a563-6db083aef051</uuid>
+  <forward mode='nat'>
+    <nat>
+      <port start='1024' end='65535'/>
+    </nat>
+  </forward>
+  <bridge name='virbr0' stp='on' delay='0'/>
+  <mac address='52:54:00:fb:7c:8e'/>
+  <ip address='192.168.122.1' netmask='255.255.255.0'>
+    <tftp root='/srv/tftpboot'/>
+    <dhcp>
+      <range start='192.168.122.2' end='192.168.122.254'/>
+      <bootp file='pxelinux.0'/>
+    </dhcp>
+  </ip>
+</network>
+```
+
+### initrd preparation
+
+Iguana provides a universal initrd in which actual functionality is implemented in containers. This
+ramdisk and its corresponding kernel are included in the [`iguana`
+package](https://build.opensuse.org/package/show/home:oholecek:iguana/iguana).
+
+Which containers to use and how to set them up is defined in a *workflow definition*. The [Iguana
+repository](https://github.com/openSUSE/iguana) includes a [definition for
+D-Installer](https://github.com/openSUSE/iguana/blob/main/iguana-workflow/examples/d-installer.yaml).
```

Comment:
Just hint, you don't have to download definition file, using raw github link as control url works too. e.g.: `rd.iguana.control_url=https://raw.githubusercontent.com/openSUSE/iguana/main/iguana-workflow/examples/d-installer.yaml`